kasa/experimental/smartcamera.py

```diff
@@ -2,31 +2,135 @@
+
+_LOGGER = logging.getLogger(__name__)
 
 
 class SmartCamera(SmartDevice):
     """Class for smart cameras."""
 
-    async def update(self, update_children: bool = False):
-        """Update the device."""
+    # Modules that are called as part of the init procedure on first update
+    FIRST_UPDATE_MODULES = {DeviceModule, ChildDevice}
+
+    @staticmethod
+    def _get_device_type_from_sysinfo(sysinfo: dict[str, Any]) -> DeviceType:
+        """Find type to be displayed as a supported device category."""
+        device_type = sysinfo["device_type"]
+        if device_type[-3:] == "HUB":
+            return DeviceType.Hub
+        return DeviceType.Camera
+
+    def _update_internal_info(self, info_resp):
+        """Update the internal device info."""
+        info = self._try_get_response(info_resp, "getDeviceInfo")
+        self._info = self._map_info(info["device_info"])
+
+    def _update_children_info(self):
+        """Update the internal child device info from the parent info."""
+        if child_info := self._try_get_response(
+            self._last_update, "getChildDeviceList", {}
+        ):
+            for info in child_info["child_device_list"]:
+                self._children[info["device_id"]]._update_internal_state(info)
+
+    async def _initialize_children(self):
+        """Initialize children for hubs."""
+        if child_info := self._try_get_response(
+            self._last_update, "getChildDeviceList", {}
+        ):
+            for info in child_info["child_device_list"]:
+                if (
+                    category := info.get("category")
+                ) and category in SmartChildDevice.CHILD_DEVICE_TYPE_MAP:
+                    child_id = info["device_id"]
+                    child_protocol = _ChildCameraProtocolWrapper(
+                        child_id, self.protocol
+                    )
+                    try:
+                        nego_response = await child_protocol.query(
+                            {"component_nego": None}
+                        )
+                        child_components = {
+                            item["id"]: item["ver_code"]
+                            for item in nego_response["component_nego"][
+                                "component_list"
+                            ]
+                        }
+                        self._children[child_id] = await SmartChildDevice.create(
+                            parent=self,
+                            child_info=info,
+                            child_components=child_components,
+                            protocol=child_protocol,
+                        )
+                    except Exception as ex:
+                        _LOGGER.exception(
+                            "Error initialising child %s: %s", child_id, ex
+                        )
+                        continue
+                    self._children[child_id]._update_internal_state(info)
```

Comment:
I'd extract this into its own method, and probably invert the checks above (i.e., bail out early if `getChildDeviceList` query isn't true-y) to keep the code flat.